doc/architecture/decisions/0009-primary-person.md

```diff
@@ -0,0 +1,69 @@
+# 9. Primary Person
+
+## Context
+
+The census requires that the primary person within a household can be identified and routed differently than other household members. Currently all people within a household would be represented as a list, with no distinction between them.
+
+## Decision
+
+### Add context to lists
+
+Context allows storage of additional information related to lists. Each list collection can store with it a map of context for the list. For example, a primary householder "Joe" would be represented as follows:
+
+```json
+{
+    "answers": [
+        {
+            "answer_id": "date-of-birth",
+            "value": "1982-08-01",
+            "list_item_id": "a9hd8j"
+        },
+        {
+            "answer_id": "first-name",
+            "value": "Joe",
+            "list_item_id": "a9hd8j"
+        },
+        {
+            "answer_id": "date-of-birth",
+            "value": "1990-05-23",
+            "list_item_id": "gyt5hy"
+        },
+        {
+            "answer_id": "first-name",
+            "value": "Jane",
+            "list_item_id": "gyt5hy"
+        },
+    ],
+    "lists": [
+        {
+            "name": "people",
+            "context": {
+                "primary_person": "a9hd8j"
+            },
+            "items": [
+                "a9hd8j",
+                "gyt5hy"
+            ]
+        }
+    ]
+}
+```
+
+### Create a PrimaryPersonCollector block type
+
+A new block type "PrimaryPersonCollector" will be defined to support the necessary behaviour. It will extend the existing ListCollector and allow addition of a primary person through adding context to the list store. The following diagram identifies the flow to be implemented in runner.  
+
+![Primary Flow](0009-primary-flow.png)
+
+Both the PrimaryPersonListCollector and ListCollector operate upon the same list in the store. The PrimaryPersonCollector would create an item in the list store, before defining any answers it refers to. This allows the identification of a primary person in one block and deferring their name entry to another. 
+
+On routing backwards from the ListCollector, a custom route handler would be implemented. This is represented in the reverse flow below, where the primary person would be unable to reach the primary name block when using the "Previous" link.
```

Comment:
This doesn't require a custom route handler if the add primary person block is defined as part of the "Do you live at?" block as the add primary person block would never be included in the routing path.